# Q3
'Kidman, Nicole', 'Radcliffe, Daniel', 'Damon, Matt', 'Allen, Tim (I)', 'Austin, Steve (IV)', 'Bonham Carter, Helena', 'Foster, Darrell', 'De Niro, Robert', 'Scorsese, Martin', 'Clooney, George'

[0.174603, 0.52, 0.119403, 0.101266, 0.205128, 0.081633, 0.102041, 0.061856, 0.102041, 0.098592]

In [3]:
library('igraph')


Attaching package: ‘igraph’

The following objects are masked from ‘package:stats’:

    decompose, spectrum

The following object is masked from ‘package:base’:

    union



In [4]:
g <- read_graph('../output/edge_list.txt', format='ncol')

In [6]:
result = sort(page_rank(g, directed = TRUE)$vector, decreasing = TRUE)

In [14]:
print(result)

       85691        27627        45392        65911        32113         6535 
1.162306e-04 1.006428e-04 8.696622e-05 8.656689e-05 8.585734e-05 8.226331e-05 
       59701        34245        20538        40331        18101        37245 
7.885718e-05 7.515654e-05 7.245503e-05 7.010544e-05 6.730552e-05 6.716240e-05 
       52758        29590        63851         7577        49596        49626 
6.648912e-05 6.522120e-05 6.494044e-05 6.400441e-05 6.391550e-05 6.321575e-05 
       64032        13285        56834        49270        38240        34427 
6.279366e-05 6.252181e-05 6.215704e-05 6.150044e-05 6.144952e-05 6.057109e-05 
       64785        66657        27085        67462        49580         3249 
5.987884e-05 5.878517e-05 5.847058e-05 5.817264e-05 5.728420e-05 5.670343e-05 
       93951        41352         8210        69872        21519        47085 
5.664631e-05 5.644248e-05 5.618940e-05 5.601549e-05 5.586371e-05 5.586183e-05 
       76450        67327        11610        61390 

In [8]:
top_10 = attr(result[1:10], 'names')

In [9]:
# print(top_10)

In [10]:
degrees = degree(g, v = V(g), mode = c("in"))
# print(degrees)

In [11]:
in_degrees = seq(0)
for (i in seq(1,length(degrees))){
    name = attr(degrees[i], "names")
    for (actor in top_10){
        if (name == actor){
            print(degrees[i])
            in_degrees = c(in_degrees, degrees[i])
        }
    }
}

32113 
 5648 
65911 
 7596 
34245 
12014 
20538 
11938 
45392 
12922 
59701 
11982 
27627 
13702 
85691 
14922 
40331 
 5200 
6535 
6418 


# Q4
## Name: #Movies, #In_degree##
Flowers, Bess: 21023, 14922  
Harris, Sam (II): 15467, 13702  
Miller, Harold (I): 14563, 12922    
Tatasciore, Fred: 10860, 7596  
Jeremy, Ron: 16630, 5648  
Blum, Steve (IX): 12472, 6418  
Sayre, Jeffrey: 11108, 11982  
Kemp, Kenner G.: 10815, 12104   
Farnum, Franklyn: 14391, 11938   
Lowenthal, Yuri: 9899, 5200  

No, they are not listed in the previous section. Because most of them are old actors/actresses and they are just not as well known as the previous actors/actresses right now.

In [15]:
in_degrees = seq(0)
famous = c(4494, 111244, 12804, 27242, 32373, 16867, 62741, 107779, 17274, 53221)
for (i in seq(1,length(result))){
    name = attr(result[i], "names")
    for (actor in famous){
        if (name == actor){
            print(result[i])
        }
    }
}
# for (i in seq(1,length(degrees))){
#     name = attr(degrees[i], "names")
#     for (actor in famous){
#         if (name == actor){
#             print(degrees[i])
#         }
#     }
# }

       27242 
3.426885e-05 
       16867 
3.393277e-05 
       53221 
2.953705e-05 
       12804 
2.754246e-05 
      107779 
2.483367e-05 
       17274 
2.469046e-05 
       62741 
2.457424e-05 
      111244 
1.692606e-05 
        4494 
1.618891e-05 
       32373 
1.120852e-05 


# Q5
## Name: Page_rank, #Movies, #In_degree 
Cruise, Tom: 1.618891e-05, 1712, 1974  
Watson, Emma (II): 1.692606e-05, 943, 866  
Clooney, George: 2.754246e-05, 1887, 3060  
Hanks, Tom: 3.426885e-05, 2101, 4014  
Johnson, Eddie (I): 1.120852e-05, 309, 1070  
Depp, Johnny: 3.393277e-05, 3132, 4152   
Smith, Will (I): 2.457424e-05, 1249, 2460  
Streep, Meryl: 2.483367e-05, 2826, 3082  
DiCaprio, Leonardo: 2.469046e-05, 1345, 2532  
Pitt, Brad: 2.953705e-05, 1786, 3406  